avocado/core/test.py

```diff
@@ -877,37 +879,50 @@ def _report(self):
             self.log.info("%s %s", self.status,
                           self.name)
 
-    def fail(self, message=None):  # pylint: disable=W0221
+    def _deprecate_params_message(func):  # pylint: disable=E0213
+        """ This decorator helps to deprecate parameter 'message' and
+        replace it with 'msg'.
+        """
+
+        @functools.wraps(func)
+        def wrapper(*args, **kwargs):
+            message = kwargs.get('message', None)
+            if message is not None:
+                kwargs['msg'] = message
+                warnings.warn("The parameter 'message' has been deprecated. Please use instead 'msg'", DeprecationWarning)
+            return func(*args, **kwargs)
+        return wrapper
+
+    @_deprecate_params_message
+    def fail(self, msg=None, **kwargs):
         """
         Fails the currently running test.
 
         After calling this method a test will be terminated and have its status
         as FAIL.
 
-        :param message: an optional message that will be recorded in the logs
-        :type message: str
-        :warning message: This parameter will changed name to "msg" in the next
-                          LTS release because of lint W0221
```

Comment:
Since we are still accepting this argument, I would still keep this warning docstring.

Reply:
I think the current warning on runtime is enough but if you insist...

Reply:
@beraldoleal I've been thinking about this. People reading the documentation are likely to never used this class before, so they'll just use msg directly. And the warning won't be helpful.
For older users, the runtime warning I added already is the way to go.

Reply:
> People reading the documentation are likely to never used this class before. And the warning won't be helpful. For older users, the runtime warning I added already is the way to go.

I see the reasoning for removing this, however this might be a strong assumption.

Current users/contributors seeing the warning message at the terminal, might need to double check the method's documentation, to understand better before adapting their code. And here we see additional information, like the reason because it is being deprecated and when it will be removed.

IMO, the same is valid for new users/contributors, since they might be looking at mixed calls (msg vs message) on examples and third-party code, and it is trying to understand what is the right way to do, without the need to run a test. So, he/she might need to double check the documentation as well.

To be clear, this is not a "strong blocker" to me, and I'm not "insisting". I just would like to make sure that our documentation is in sync with our code.

Perhaps, I was not so clear the last time we spoke, and I apologize for that. But, maybe this will not be needed at all, as I tried to say before. So possibly, one way is just hold this commit until we double check the real need for this decorator vs "we are safe to remove it since we are not breaking our promise regarding the API compatibly.

Reply:
@beraldoleal I see.  When I added the decorator was with the expectation this PR could be merged fast.
 I think it's a very good solution  (*) because it handles both variable names and it doesn't break anything. It's a pity we didn't handle things this way from the beginning, but it's never late, we could keep this decorator and remove it after the next LTS.

(*) I expect almost a full day on this, so I'm clearly biased here!